test/unit/BMath.t.sol

```diff
@@ -4,564 +4,263 @@ pragma solidity 0.8.25;
+    _weightIn = bound(_weightIn, BONE, type(uint256).max);
+    _balanceIn = bound(_balanceIn, (type(uint256).max - weightIn / 2), type(uint256).max);
 
-    // Post condition
-    // it will revert (overflow)
-    //     token balance in * BONE + (token weight in/2) > uint max
-    vm.expectRevert('ERR_DIV_INTERNAL');
+    // it should revert
+    //     bi * BONE + (wi / 2) > uint256 max
+    vm.expectRevert(BNum.BNum_DivInternal.selector);
 
-    // Action
-    bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
+    bMath.calcSpotPrice(_balanceIn, _weightIn, balanceOut, weightOut, swapFee);
   }
 
-  function test_CalcSpotPriceWhenWeightedTokenBalanceOutOverflows(
-    uint256 _tokenBalanceOut,
-    uint256 _tokenWeightOut
-  ) external {
-    // Precondition
-    uint256 _tokenWeightIn = BONE;
-    uint256 _tokenBalanceIn = BONE;
-    uint256 _swapFee = BONE / 2;
-
-    _tokenWeightOut = bound(_tokenWeightOut, BONE, type(uint256).max);
-    _tokenBalanceOut = bound(_tokenBalanceOut, (type(uint256).max - _tokenWeightOut / 2), type(uint256).max);
+  function test_CalcSpotPriceWhenWeightedTokenBalanceOutOverflows(uint256 _balanceOut, uint256 _weightOut) external {
+    _weightOut = bound(_weightOut, BONE, type(uint256).max);
+    _balanceOut = bound(_balanceOut, (type(uint256).max - weightOut / 2), type(uint256).max);
 
-    // Post condition
-    // it will revert (overflow)
-    //     token balance out * BONE + (token weight out/2) > uint max
-    vm.expectRevert('ERR_DIV_INTERNAL');
+    // it should revert
+    //     bo * BONE + (wo / 2) > uint256 max
+    vm.expectRevert(BNum.BNum_DivInternal.selector);
 
-    // Action
-    bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
+    bMath.calcSpotPrice(balanceIn, weightIn, _balanceOut, _weightOut, swapFee);
   }
 
   function test_CalcSpotPriceWhenUsingASwapFeeOfZero() external {
-    // Precondition
+    // it should return correct value
+    //     bi/wi * wo/bo
+    //     20/1 * 2/30 = 1.333333...
     uint256 _swapFee = 0;
 
-    uint256 _tokenWeightIn = 10 * BONE;
-    uint256 _tokenWeightOut = 20 * BONE;
-    uint256 _tokenBalanceIn = 100 * BONE;
-    uint256 _tokenBalanceOut = 30 * BONE;
+    uint256 _spotPrice = bMath.calcSpotPrice(balanceIn, weightIn, balanceOut, weightOut, _swapFee);
 
-    // expected is bi/wi / bo/wo
-    uint256 _expected = bNum.bdivExposed(
-      bNum.bdivExposed(_tokenBalanceIn, _tokenWeightIn), bNum.bdivExposed(_tokenBalanceOut, _tokenWeightOut)
-    );
-
-    // Action
-    uint256 _spotPrice =
-      bMath.calcSpotPrice(_tokenBalanceIn, _tokenWeightIn, _tokenBalanceOut, _tokenWeightOut, _swapFee);
-
-    // Post condition
-    // it should return bi/wi / bo/wo
-    assertEq(_spotPrice, _expected, 'wrong spot price when fee is 0');
+    assertEq(_spotPrice, 1.333333333333333333e18);
```

Comment:
I would be okay with a ranged assert as well

Reply:
i think it doesn't hurt, as it forces the dev to look at the value, copy and paste it and make sure it makes sense, we had some cases in which the tolerance ranges weren't properly set and the test would PASS anyways